Sources/StreamChat/StateLayer/ChatList.swift

```diff
@@ -0,0 +1,99 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// An object which represents a list of `ChatChannel`.
+@available(iOS 13.0, *)
+public struct ChatList {
+    /// The query specifying and filtering the list of channels.
+    public let query: ChannelListQuery
+    
+    private let channelListQueryLinkingCoordinator: ChannelListQueryLinkingCoordinator
+    private let channelListUpdater: ChannelListUpdater
+    private let paginatedChannelsLoader: PaginatedChannelsLoader
+    
+    init(channels: [ChatChannel], query: ChannelListQuery, dynamicFilter: ((ChatChannel) -> Bool)?, channelListUpdater: ChannelListUpdater, client: ChatClient, environment: Environment = .init()) {
+        self.channelListUpdater = channelListUpdater
+        self.query = query
+        let state = environment.stateBuilder(
+            channels,
+            query,
+            client.config,
+            client.databaseContainer
+        )
+        channelListQueryLinkingCoordinator = environment.channelListQueryLinkingCoordinatorBuilder(
+            state,
+            query,
+            dynamicFilter,
+            channelListUpdater,
+            client.eventsController(),
+            client.databaseContainer,
+            client.config
+        )
+        paginatedChannelsLoader = environment.paginatedChannelsLoaderBuilder(
+            query,
+            client.config,
+            channelListUpdater,
+            client.channelRepository
+        )
+        self.state = state
+    }
+    
+    /// An observable object representing the current state of the channel list.
+    public let state: ChatListState
+    
+    // MARK: - Channel List Pagination
+    
+    /// Loads channels for the specified pagination parameters and updates ``ChatListState.channels``.
+    ///
+    /// - Parameters:
```

Comment:
That is true, I ended up manually changing it since then it is consistent with other parameter formats. If you prefer that, I have no problem for changing the format to that. There are probably many other cases as well. Adding an item to the todo list in Notion.